### PR TITLE
Split Linearize into two modules

### PR DESCRIPTION
**Pull request by gretay-js**
This PR moves type declarations from Linearize into their own module, called Linear. It separates the description of the IR from transformations performed on it, similarly to the way Cmm and Mach are already arranged. There is no change in functionality.

### CODE REVIEW

File: asmcomp/linearize.ml [hunk 1]
```diff
@@ -308,7 +234,7 @@ let rec linear i n =
       n3
   | Iexit nfail ->
       let lbl, t = find_exit_label_try_depth nfail in
-      assert (i.Mach.next.desc = Iend);
+      assert (i.Mach.next.desc = Mach.Iend);
```
Comment (dra27) [inline asmcomp/linearize.ml:237]: Is this strictly necessary?

Comment (gretay-js) [reply asmcomp/linearize.ml:237]: Without it, I get
``` Error (warning 18): this type-based constructor disambiguation is not principal. ```
Is there another way to fix it?

Comment (dra27) [reply asmcomp/linearize.ml:237]: Oh, it's because of the loss of the `open Mach` at the top - sorry for the noise!

**Comment by gretay-js** [comment]
I've just rebased this simple PR.

Linearize is a special case, when the IR and its construction are defined in the same module, whereas language definitions of Mach, Cmm, and other IRs are separate from modules that manipulate them. This PR brings Linearize in line with the structure of other passes.

@xavierleroy are you happy with this change?

**Review by gasche — APPROVED** [review]
This is obviously a reasonable chance to make. Approved.

P.S.: the `signals.h` dependency in `.depend` seems both unrelated and correct. I think it's slightly nicer to have a separate commit with just `make alldepend` when trunk is not dependency-correct (which is often, unfortunately).

**Comment by gretay-js** [comment]
Thank you!  
Sorry about muddling .depend. I'll make sure to separate unrelated .depend changes in the future.

**Pull request merged**